### PR TITLE
Refactor frontend

### LINKED ISSUE #65: Separate Film.vue into proper components

### PR DESCRIPTION
**Pull request by qjack001**
I'm going to finally do it. I'll try and use the new advanced-search features as a motivator to finish the refactor first. 

This PR is a work in progress, I'll update it as I go along, so you can review it component-by-component rather than all at once (feel free to join in, if you want). Closes #65

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,41 +4,18 @@
...
 	"devDependencies": {
 		"@vue/cli-plugin-babel": "~4.5.0",
-		"@vue/cli-plugin-eslint": "~4.5.0",
 		"@vue/cli-service": "~4.5.0",
-		"babel-eslint": "^10.1.0",
-		"eslint": "^6.7.2",
-		"eslint-plugin-vue": "^6.2.2",
 		"vue-template-compiler": "^2.6.11"
 	},
-	"eslintConfig": {
```
Comment (qjack001) [inline package.json:23]: I'm sorry Quinn, your linter is just too agressive for me.

File: src/main.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import Vue from 'vue'
 import App from './App.vue'
+import FocusVisible from 'vue-focus-visible'
```
Comment (qjack001) [inline src/main.js:3]: This is the thing that I used on Bookcase to polyfill `:focus-visible`

File: package.json (hunk 1 above)
Comment (HoloPollock) [reply package.json:23]: Fair I agree lol it's the default I don't love it thought

File: src/App.vue [hunk 3]
```diff
@@ -28,8 +31,22 @@
...
+		{
+			return {
+				users: ''  
+			}
+		},
+		methods:
+		{
+			submit()
+			{
+				alert(this.users)
```
Comment (qjack001) [inline src/App.vue:47]: Just a placeholder for now.

**Comment by qjack001** [comment]
Basically, all the logic has been moved into `App.vue`, which also handles high level layout and styling (colour schemes, etc). Each component contains its own styling, and as little logic as nessesary (i.e. the `SearchBar` component doesnt actually do the search, it just handles the input). Still to-do (but in a future PR): rethink how the footer works, refactor the darkmode component, and re-check accessibility.

One other thing: I turned-off the `ignore` url-param. I have an idea for how it'll work with the short-film/feature-length options when we get there.

**Review by maxkele — APPROVED** [review]

**Comment by qjack001** [comment]
Ok I lied, I did the footer and darkmode refactors. Now closes #83

**Review by HoloPollock — APPROVED** [review]
great work

File: src/components/SearchBar.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	{
+		name: 'SearchBar',
+		props:
+		[
+			'value',
+			'action'
+		],
+		methods:
+		{
+			updateValue: function (value)
```
Comment (HoloPollock) [inline src/components/SearchBar.vue:35]: this looks weird?

File: src/components/SearchBar.vue [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+			placeholder="ex: holopollock, qjack"
+			:value="value"
+			v-on:keyup.enter=action()
+			v-on:input="updateValue($event.target.value)"
+		/>
+		<button 
+			v-on:click=action()
+		>
+			Submit
+		</button>
```
Comment (HoloPollock) [inline src/components/SearchBar.vue:19]: any ides why this block syntax highlighting?

**Comment by HoloPollock** [comment]
one day you should teach me how emits work

File: src/components/SearchBar.vue (hunk 4 above)
Comment (qjack001) [reply src/components/SearchBar.vue:35]: like the `name: function (value)`-formatting? It was just the way they did it in the Vue docs, I can change it if you'd prefer.

**Pull request merged**